core/templates/domain/objects/NumberWithUnitsObjectFactory.ts

```diff
@@ -123,6 +136,40 @@ export class NumberWithUnits {
       units: this.units,
     };
   }
+
+  getCanonicalRepresentationOfUnits(): Unit[] {
+    const updatedUnits = this.units.map(({unit, exponent}: Unit) => {
+      const baseUnit = currencyUnits.includes(unit)
+        ? ObjectsDomainConstants.UNIT_TO_NORMALIZED_UNIT_MAPPING[unit]
+        : mathjsUnit(unit).units[0].unit.name;
+
+      const unitPrefix = currencyUnits.includes(unit)
+        ? ''
+        : mathjsUnit(unit).units[0].prefix.name;
+      let normalizedUnit = '';
```

Comment:
We can't use this directly for all the cases as we have to avoid calling mathjs's unit() function for currency units. and sure I will try to make it simpler.

Reply:
Er, to be clear, I'm suggesting this as a replacement to lines 149-158, not the part before it. PTAL again and see what you think.

Reply:
Oh that makes sense, Done.